### ISSUE TITLE
No alt attribute for thumbnails navigation

### ISSUE THREAD

**Issue by cgrymala** [woothemes/FlexSlider#1389]
For accessibility purposes, all images are *required* to have an alt attribute, even if it's empty. Unfortunately, the `<img>` tags that are generated for the thumbnails navigation do not include the alt attribute.

I would recommend the following changes:

1. Allow a data-thumbnail-alt attribute on the slides themselves, and use that as the text of the alt attribute on the thumbnails if it exists.
1. If the data-thumbnail-alt attribute does not exist, apply an empty alt attribute to the thumbnails `<img>` tags.

A suggestion would be, on [line 220 of jquery.flexslider.js](https://github.com/woothemes/FlexSlider/blob/8ed61acca0a1c1f1242ebb0cdb8291be6d4a0d91/jquery.flexslider.js#L220), adjust the code:

```javascript
              item = (slider.vars.controlNav === "thumbnails") ? '<img src="' + slide.attr( 'data-thumb' ) + '"/>' : '<a>' + j + '</a>';
```

 so that it looks something like:

```javascript
              if ( undefined == slide.attr( 'data-thumb-alt' ) ) {
                slide.attr( 'data-thumb-alt', '' );
              }
              item = (slider.vars.controlNav === "thumbnails") ? '<img src="' + slide.attr( 'data-thumb' ) + '" alt="' + slide.attr( 'data-thumb-alt' ) + '"/>' : '<a>' + j + '</a>';
```

Pull request incoming. Thank you.

**Comment by rjh123** [comment]
I've followed the instructions in the commit and my thumbnail nav images still don't have alt text on them. I have the most recent version of the Drupal module (7.x-2.0-rc1) and the latest Flexslider code (2.6.0). Is anyone else having this problem? I'd like some ideas how to solve this as my accessibility score is too low. Any help appreciated.

**Comment by cgrymala** [comment]
@rjh123 The way they've implemented it in the final product, the `data-thumb-alt` attribute of the slide itself needs to be set to something other than `''`. It would have been nice if they'd implemented it to support empty alt attributes, as alt attributes are required, whether they have text or not; but that's not the way it's currently implemented.

Therefore, make sure that the individual slides (not the thumbnails, but the normal slides) have a `data-thumb-alt` attribute, and that attribute has some content.

An example might look like:

```html
<li class="slide" data-thumb="//www.example.com/thumb.jpg" data-thumb-alt="My thumbnail image alt attribute"><img src="//www.example.com/image.jpg" alt="My main image"/></li>
```

**Comment by rjh123** [comment]
(Working through the Drupal interface) I don't see a way to put data-thumb attributes in the slides without  writing php. But from what you're saying, I need to just get the data-thumb-alt attribute on my image-wrapping element, and Flexslider will put the alt attribute with alt text on the navigation image thumbs. Is that correct?

**Comment by rjh123** [comment]
Oops! I just reread your post and realize my last comment won't make sense. I still think Flexslider allows me to do this without some custom php. But I'll look into it further, thanks.

**Comment by nohart** [comment]
Has this issue ever been fixed? I am running Magento v1.9.3.7 with a BuyShop Retina theme installed. It has the Flex Slider and it doesn't allow for alt or title tags. Does anyone know how to fix this?

**Comment by laptopleon** [comment]
I'm having the same problem: No alt tags are rendered on images from the FlexSlider, in my case in Wordpress WooCommerce Frontstore Product page image gallery. Even though they are in the HTML, they don't show up on the screen or in the browser code inspector. Annoying for people who can't see well. Can't be good for SEO either, because Google bots look further than just the raw code.

Can this please be fixed? How can I contribute?
